### PR TITLE
s2i bump

### PR DESCRIPTION
**Pull request by bparees**
@jim-minter ptal

### CODE REVIEW

**Comment by bparees** [comment]
[testextended][extended:core(builds|image_ecosystem)]

**Comment by bparees** [comment]
[test]

**Comment by bparees** [comment]
@jim-minter assuming this passes tests now, ptal.  particularly where i had to change to use CreateTarStream instead of StreamFileAsTarWithCallback

File: pkg/bootstrap/docker/dockerhelper/filetransfer.go [hunk 1]
```diff
@@ -116,18 +117,16 @@ func DownloadDirFromContainer(client *docker.Client, container, src, dst string)
...
 // UploadFileToContainer uploads a file to a remote container.
 func UploadFileToContainer(client *docker.Client, container, src, dest string) error {
 	uploader, errch := newContainerUploader(client, container, filepath.Dir(dest))
 
-	nullWalkFunc := func(path string, info os.FileInfo, err error) error { return err }
-
-	t := stitar.New()
-	err := t.StreamFileAsTarWithCallback(src, filepath.Base(dest), uploader, nullWalkFunc, false)
+	t := s2itar.New(s2iutil.NewFileSystem())
+	err := t.CreateTarStream(src, false, uploader)
```
Comment (jim-minter) [inline pkg/bootstrap/docker/dockerhelper/filetransfer.go]:
```go
	err := t.CreateTarStream(src, false, uploader)
```

should read:

```go
	tarWriter := s2itar.RenameAdapter{Writer: tar.NewWriter(uploader), Old: filepath.Base(src), New: filepath.Base(dest)}

	err := t.CreateTarStreamToTarWriter(src, true, tarWriter, nil)
	if err == nil {
		err = tarWriter.Close()
	}

```

**Comment by jim-minter** [comment]
@bparees bar the above change, looks fine to me.

File: pkg/bootstrap/docker/dockerhelper/filetransfer.go (hunk 1 above)
Comment (bparees) [inline pkg/bootstrap/docker/dockerhelper/filetransfer.go]: only close tarWriter if err is nil?

Comment (jim-minter) [inline pkg/bootstrap/docker/dockerhelper/filetransfer.go]: it's the best I could think of - tarWriter.Close() sends the terminating '\0's record file which are supposed to indicate to tar that its stream isn't erroneously truncated.

Comment (jim-minter) [inline pkg/bootstrap/docker/dockerhelper/filetransfer.go]: that code is a copy/paste of the equivalent that I implemented in s2i

**Comment by bparees** [comment]
@jim-minter it seems plausible the latest s2i changes broke extended builds in origin... (see latest extended test failures)

**Comment by bparees** [comment]
```
2016-12-14T15:47:03.563740000Z I1214 15:47:03.562961       1 postexecutorstep.go:322] Uploading directory "/tmp/s2i-build213899081/upload/runtimeArtifacts" -> "/opt/jboss"
2016-12-14T15:47:03.564150000Z I1214 15:47:03.562992       1 docker.go:390] Uploading "/tmp/s2i-build213899081/upload/runtimeArtifacts" to "/opt" ...
2016-12-14T15:47:03.564468000Z I1214 15:47:03.563121       1 tar.go:199] Adding "/tmp/s2i-build213899081/upload/runtimeArtifacts" to tar ...
2016-12-14T15:47:03.564871000Z I1214 15:47:03.563441       1 tar.go:285] Adding to tar: /tmp/s2i-build213899081/upload/runtimeArtifacts/hello.jar as runtimeArtifacts/hello.jar
2016-12-14T15:47:03.565371000Z I1214 15:47:03.563711       1 tar.go:285] Adding to tar: /tmp/s2i-build213899081/upload/runtimeArtifacts/scripts/assemble-runtime as runtimeArtifacts/scripts/assemble-runtime
2016-12-14T15:47:03.565577000Z I1214 15:47:03.563926       1 tar.go:285] Adding to tar: /tmp/s2i-build213899081/upload/runtimeArtifacts/scripts/run as runtimeArtifacts/scripts/run
2016-12-14T15:47:03.603253000Z I1214 15:47:03.602678       1 postexecutorstep.go:328] Uploading file "/tmp/s2i-upload-done050185315" -> "/tmp/s2i-upload-done050185315"
2016-12-14T15:47:03.603597000Z I1214 15:47:03.602711       1 docker.go:390] Uploading "/tmp/s2i-upload-done050185315" to "/tmp" ...
2016-12-14T15:47:03.603925000Z I1214 15:47:03.602844       1 tar.go:199] Adding "/tmp/s2i-upload-done050185315" to tar ...
2016-12-14T15:47:03.604548000Z I1214 15:47:03.602893       1 tar.go:285] Adding to tar: /tmp/s2i-upload-done050185315 as s2i-upload-done050185315
2016-12-14T15:47:04.066532000Z I1214 15:47:04.065901       1 postexecutorstep.go:340] /bin/sh: /opt/jboss/scripts/assemble-runtime: Permission denied
```

seems like the permissions aren't getting set properly on the assemble-runtime script... investigating.

**Comment by bparees** [comment]
@php-coder it looks like the scripts in your test repo are not marked executable:
https://github.com/php-coder/java-maven-hello-world/tree/s2i-assemble-and-assemble-runtime/.s2i/bin

I guess this used to work, but would you expect it to work?

**Comment by bparees** [comment]
prereq PRs (I think):
https://github.com/php-coder/java-maven-hello-world/pull/2
https://github.com/php-coder/java-maven-hello-world/pull/1

**Comment by php-coder** [comment]
@bparees Thank you, Ben! It works before but now I don't know how :-/

**Comment by bparees** [comment]
I think our Tar logic may have been updating the permissions.

**Comment by bparees** [comment]
still failing :( 
postexecutorstep.go:340] /bin/sh: /opt/jboss/scripts/assemble-runtime: Permission denied

I assume it has to do w/ this change in s2i:
https://github.com/openshift/source-to-image/commit/c0eb0e35ed044c0c9319e3978e77a7c10b47070f#diff-6b4b95c3fb2a650ac8b9a0c5eabdb8b3L129

And that we are now not going through the same chmod path.

**Comment by jim-minter** [comment]
My bad.  I'm hoping that https://github.com/openshift/source-to-image/pull/662 will fix this, but I'm still testing it here.

**Comment by jim-minter** [comment]
Confirmed, with this PR + https://github.com/openshift/source-to-image/pull/662, the extended case 'should use assemble-runtime script from URL' now works for me.

@php-coder I don't mind either way but I would be tempted to revert https://github.com/php-coder/java-maven-hello-world/pull/1 and https://github.com/php-coder/java-maven-hello-world/pull/2, FWIW.

**Comment by php-coder** [comment]
@jim-minter You want to revert them just to make sure that your fix is working?

**Comment by bparees** [comment]
@php-coder shouldn't be necessary, since we're still broken right now (pending @jim-minter's changes), so we should already know if his changes fix it.

**Comment by php-coder** [comment]
@bparees Ok. Just in case I invite you and @jim-minter to be collaborators in this repo, so you will be able to do any changes that may be required. I don't want to block you.

**Comment by bparees** [comment]
@php-coder thanks!

**Comment by bparees** [comment]
flake https://github.com/openshift/origin/issues/10663
[test]

**Comment by bparees** [comment]
[merge]

Ben Parees | OpenShift

**Comment by bparees** [comment]
flake https://github.com/openshift/origin/issues/9886
[merge]

**Pull request merged**